### ISSUE TITLE
segfault on pikepdf.open(bin_stream) for some pdf's

### ISSUE THREAD

**Issue by wv0m56** [pikepdf/pikepdf#10]
Failed with segfault:
```
f = open("Accessible EPUB 3.pdf", "rb")
doc = pikepdf.open(f)
for page in doc.pages:
    page.Rotate = 180
doc.save('test-rotated.pdf')
```

OK:
```
doc = pikepdf.open("Accessible EPUB 3.pdf")
for page in doc.pages:
    page.Rotate = 180
doc.save('test-rotated.pdf')
```

Other tests:
binary stream open (first snippet above) on tests/resources/pal-1bit-trivial.pdf -- OK (even though result is just a blank page)
my ebook collection |> "PathLike" open (second) snippet |> binary stream open snippet -- OK
my ebook collection |> binary stream open -- failed
my ebook collection |> various qpdf cli commands -- OK

On another note, the front page mentions "Commercial support is available" but I can't find any info about this anywhere. Do you have a pointer on this?

Thanks

**Comment by jbarlow83** [comment]
Thank you for this report. Any segfault is a serious concern.

I can't reproduce it but I'm fairly sure it's a memory use-after-free issue. In the interest of reproducing, what platform are you on, and are you using pikepdf from a wheel or did you build a version?

Regarding commercial support - contact me at jim@purplerock.ca and tell me a little about your project and requirements.

**Comment by wv0m56** [comment]
IIRC I installed using pip.
I can reproduce the problem using the following: http://www.planetpdf.com/planetpdf/pdfs/pdf2k/01W/rosenthol_intro2pdfprog.pdf

Platform:
```
$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
```

```
$ pip3 list | grep pikepdf
pikepdf                       0.3.4
```

```
$ python3 -V
Python 3.6.6
```

Thank you for the info on commercial support. I'm still evaluating my options so no promises. I'll email you if I decide to go further in this direction.

**Comment by jbarlow83** [comment]
Should be fixed in v0.3.5/master... still can't reproduce, but I'm pretty sure I fixed it.

**Comment by wv0m56** [comment]
Since you can't reproduce it, I just want to let you know that the fix works. Thanks a lot!
(I hope I don't reopen the issue)
